core/src/main/java/com/nightscout/core/barcode/NSBarcodeConfig.java

```diff
@@ -69,6 +69,20 @@ public void configureBarcode(String jsonConfig) {
         return apiUris;
     }
 
+    public Optional<String> getMqttUri() {
+        String mqttUri = null;
+        try {
+            if (hasMqttConfig()) {
+                mqttUri = config.getJSONObject(NSBarcodeConfigKeys.MQTT_CONFIG).getString(NSBarcodeConfigKeys.MQTT_URI);
+            } else {
```

Comment:
you can remove this and the return in the `catch` block, and just return Optional.fromNullable(mqttUri) on line 83.